packages/bierzo-wallet/src/routes/receivePayments/index.tsx

```diff
@@ -17,38 +17,34 @@ function onReturnToBalance(): void {
 const ReceivePayment = (): JSX.Element => {
   const pubKeys = ReactRedux.useSelector((state: RootState) => state.extension.keys);
 
-  const [chainAddressMap, setChainAddressMap] = React.useState<ChainAddressMap[]>([]);
+  const [chainAddresses, setChainAddresses] = React.useState<ReadonlyArray<ChainAddress>>([]);
 
   React.useEffect(() => {
     async function processAddresses(pubKeys: { [chain: string]: string }): Promise<void> {
-      const addressesMap: ChainAddressMap[] = [];
-      const chainIds = Object.keys(pubKeys);
-      for (let i = 0; i < chainIds.length; i++) {
-        const chainId = chainIds[i] as ChainId;
-        const plainPubkey = pubKeys[chainId];
-        if (!plainPubkey) {
-          return;
-        }
-
-        const identity: Identity = TransactionEncoder.fromJson(JSON.parse(plainPubkey));
-        const address = (await getCodecForChainId(chainId)).identityToAddress(identity);
-        const chainName = getChainName(chainId);
-        addressesMap.push({
-          chainId: chainId,
-          chainName,
-          address,
+      const identities = Object.values(pubKeys).map(
```

Comment:
If you will get me code and remove unnecessary lines you will get it at least 6 lines shorter, without unnecessary `.map` loop.

Reply:
Yeah, I'm generally not so happy with all thise raw pubkey maps in the codebase. It should be even easier to get the list of identities from redux. But one step at a time